src/Type/Accessory/HasOffsetValueType.php

```diff
@@ -122,6 +122,10 @@ public function isSubTypeOf(Type $otherType): IsSuperTypeOfResult
 
 	public function isAcceptedBy(Type $acceptingType, bool $strictTypes): AcceptsResult
 	{
+		if ($acceptingType->isArray()->yes() && $acceptingType->isIterableAtLeastOnce()->yes()) {
+			return AcceptsResult::createYes();
```

Comment:
It should check the iterable value type against `$this->valueType`.

Reply:
I can't check the iteratable-value-type and the non-emptiness in a single IF because `$acceptingType` is invoked once for each element of the intersection.

this means I get the `ArrayType` which I could check against the value-type, but not against non-emptiness
and on the 2nd call I get the `NonEmptyArray` type, which I can check only against the non-emptiness, but not the value type

Reply:
I will think more about it.. maybe we need a fix directly on `IntersectionType` then